management_app/templates/faculty/faculty-dashboard.html

```diff
@@ -0,0 +1,87 @@
+{% extends 'base.html' %}
+
+{% block title %}Faculty{% endblock %}
+{% block header %}Hi {{ g.user['user_name'] }}{% endblock %}
+
+{% block content %}
+<div class="col-3 mb-3">
+  <form class="d-flex">
+    <label class="align-self-center" for="year">Year:&nbsp;</label>
+    <select class="form-select align-self-center" id="year" aria-label="Year Select" name="year">
+      {% for option in year_options %}
+      <option
+        value="{{ option }}" {% if request.args.get('year') == option or request.args.get('year') == None %} selected="selected"{% endif %}>
+        {{ option }}
+      </option>
+      {% endfor %}
+    </select>&nbsp;
+    <button type="submit" class="btn btn-outline-primary" id="submitFilterByYear">Filter</button>
+   </form>
+</div>
+
+<div style="font-size: 18px;">
+  {% if point_info %}
+  <h5><b>Previous year ending balance:</b> {{ point_info['previous_balance'] }}</h5>
+  <h5><b>Ending balance:</b> {{ point_info['ending_balance'] }}</h5>
+
+  <ul class="mb-3">
+    <li>Current credit due: {{ point_info['credit_due'] }}</li>
+    <li>Teaching points: {{ point_info['teaching_point'] }}</li>
+    <li>Grad points: {{ point_info['grad_point'] }}
+      <ul>
+        <li>Grad Count: {{ point_info['grad_count'] }}</li>
+        {% if point_info['grad_students']|length > 0 %}
```

Comment:
This throws an error if the grad students column in the faculty_point_info table is null. Although if it's an empty string then it works as expected.

@yuchesu If the grad count is zero meaning that the faculty member does not have any grad students, should the grad students column in the faculty_point_info table be null or an empty string?

Reply:
Just found some issues as well. Will work on the fix later.

Reply:
Updated the logic for grad students.
- If grad_students input from upload professor_point_info spreadsheet file is empty, then store null.
- If grad_students input from UI is null, also store null.